pkg/codegen/testing/test/testdata/assets-and-archives/docs/resourcewithassets/_index.md

```diff
@@ -14,8 +14,173 @@ no_edit_this_page: true
 
 
 
+## Creating a ResourceWithAssets Resource
+<div>
+<pulumi-chooser type="language" options="typescript,python,go,csharp,java,yaml"></pulumi-chooser>
+</div>
+
+<div>
+    <pulumi-choosable type="language" values="javascript,typescript">
+        <div class="highlight">
+            <pre class="chroma">
+                <code class="language-typescript" data-lang="typescript">
+import * as pulumi from "@pulumi/pulumi";
+import * as example from "@pulumi/example";
+
+const resourceWithAssets = new example.ResourceWithAssets("resourceWithAssets", {
+  archive: new pulumi.asset.FileAsset("./file.txt"),
+  nested: {
+    archive: new pulumi.asset.FileAsset("./file.txt"),
+    asset: new pulumi.asset.StringAsset("Hello, world!"),
+    plainArchive: new pulumi.asset.FileAsset("./file.txt"),
+    plainAsset: new pulumi.asset.StringAsset("Hello, world!"),
+  },
+  source: new pulumi.asset.StringAsset("Hello, world!"),
+});
+
+                </code>
+            </pre>
+        </div>
+    </pulumi-choosable>
+</div>
+
+<div>
+    <pulumi-choosable type="language" values="python">
+        <div class="highlight">
+            <pre class="chroma">
+                <code class="language-python" data-lang="python">
+import pulumi
+import pulumi_example as example
+
+resourceWithAssets = example.ResourceWithAssets("resourceWithAssets",
+  archive=pulumi.FileAsset("./file.txt"),
+  nested=example.TypeWithAssetsArgs(
+    archive=pulumi.FileAsset("./file.txt"),
+    asset=pulumi.StringAsset("Hello, world!"),
+    plain_archive=pulumi.FileAsset("./file.txt"),
+    plain_asset=pulumi.StringAsset("Hello, world!"),
+  ),
+  source=pulumi.StringAsset("Hello, world!")
+)
+
+                </code>
+            </pre>
+        </div>
+    </pulumi-choosable>
+</div>
+
+<div>
+    <pulumi-choosable type="language" values="csharp">
+        <div class="highlight">
+            <pre class="chroma">
+                <code class="language-csharp" data-lang="csharp">
+using Pulumi;
+using Example = Pulumi.Example;
+
+var resourceWithAssets = new Example.ResourceWithAssets("resourceWithAssets", new () 
+{
+  Archive = new FileAsset("./file.txt"),
+  Nested = new Example.Inputs.TypeWithAssetsArgs
+  {
+    Archive = new FileAsset("./file.txt"),
+    Asset = new StringAsset("Hello, world!"),
+    PlainArchive = new FileAsset("./file.txt"),
+    PlainAsset = new StringAsset("Hello, world!"),
+  },
+  Source = new StringAsset("Hello, world!"),
+});
+
+                </code>
+            </pre>
+        </div>
+    </pulumi-choosable>
+</div>
+
+<div>
+    <pulumi-choosable type="language" values="go">
+        <div class="highlight">
+            <pre class="chroma">
+                <code class="language-go" data-lang="go">
+import (
+  "github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+  "assets-and-archives/example/index"
+)
+
+resourceWithAssets, err := example.NewResourceWithAssets("resourceWithAssets", &example.ResourceWithAssetsArgs{
```

Comment:
This import doesn't match.

Reply:
Here I am using the import base path provided from the go language options in the schema. Maybe I shouldn't do that and instead just follow the convention of `pulumi/pulumi-{pkg}/...` etc. 🤔

Reply:
There must be a function that handles this correctly already, so I would look for that.

In this case, `index` is special, so `../example/index` should be imported as `../example`. 

`example.NewResourceWithAssets` is correct. `"assets-and-archives/example/index"
` is not.